### PR TITLE
Add tooltips for input fields

### PR DESCRIPTION
**Pull request by svpgaicean**

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ export const inputValidationRegex: IOptions = {
 	username: /^[a-zA-Z\d.]{6,}$/,
 	password: /^(?!.*?[\^])(?=.*[a-z])(?=.*[A-Z])(?=.*\d)[A-Z]+.{7,}$/,
 	confirmed: /^(?!.*?[\^])(?=.*[a-z])(?=.*[A-Z])(?=.*\d)[A-Z]+.{7,}$/,
-	email: /^.*$/
+	email: /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
 };
 
 export class Authentication {
```
Comment (SVARusu) [inline src/utils.ts]: You might get some errors if you export the class like that

File: src/registration.ts [hunk 2]
```diff
@@ -46,20 +51,10 @@ class Registration extends Authentication {
 const user = new Registration();
 
 export function registerNewUser() {
```
Comment (SVARusu) [inline src/registration.ts]: Just do your exports at the bottom of the file >_<

File: src/utils.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (svpgaicean) [reply src/utils.ts]: moved exports at bottom of file

File: src/registration.ts (hunk 2 above)
Comment (svpgaicean) [reply src/registration.ts]: moved exports at bottom of file.

File: tsconfig.json [hunk 3]
```diff
@@ -5,7 +5,7 @@
   "compilerOptions": {
     /* Basic Options */
     // "incremental": true,                   /* Enable incremental compilation */
-    "target": "es5",                          /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019' or 'ESNEXT'. */
+    "target": "es2019",                          /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019' or 'ESNEXT'. */
```
Comment (SVARusu) [inline tsconfig.json]: why 2019

File: src/index.ts [hunk 4]
```diff
@@ -1,3 +1,3 @@
 import { registerNewUser } from './registration';
 
-registerNewUser();
\ No newline at end of file
+registerNewUser().then((user) => { console.log(user) });
```
Comment (SVARusu) [inline src/index.ts]: what's the point of this?

File: src/registration.ts [hunk 5]
```diff
@@ -39,28 +39,31 @@ class Registration extends Authentication {
...
 			username: this._username,
 			password: this._password,
-			email: this._email
+      email: this._email,
+      deleted: false
 		};
 	};
 }
 
 const user = new Registration();
```
Comment (SVARusu) [inline src/registration.ts]: group your variable declarations

File: tsconfig.json (hunk 3 above)
Comment (svpgaicean) [reply tsconfig.json]: because we are in 2019.

File: src/index.ts (hunk 4 above)
Comment (svpgaicean) [reply src/index.ts]: to use the user data for other stuff.

File: src/registration.ts (hunk 5 above)
Comment (svpgaicean) [reply src/registration.ts]: can't group variable declarations if you only have one variable declaration *Roll Safe*

**Pull request merged**